renderapi/client/client_calls.py

```diff
@@ -228,12 +264,44 @@ def tilePairClient(stack, minz, maxz, outjson=None, delete_json=False,
                        subprocess_mode=subprocess_mode,
                        add_args=argvs, **kwargs)
 
-    with open(outjson, 'r') as f:
-        jsondata = json.load(f)
+    # We create the jsonfile, so if it is empty it could be multiple
+    if os.stat(outjson).st_size == 0:
+        # outjson we created is empty, so remove it
+        os.remove(outjson)
+
+        outbn_root, outbn_ext = os.path.splitext(os.path.basename(outjson))
+        outbn_pattern = "{root}_p[0-9]+{ext}".format(
+            root=outbn_root, ext=outbn_ext)
+        jsonfiles = [
+            os.path.join(os.path.dirname(outjson), bn)
+            for bn in os.listdir(os.path.dirname(outjson))
+            if re.match(outbn_pattern, bn)]
+    else:
+        jsonfiles = [outjson]
+
+    if return_jsondata:
+        jsondata_list = []
+        for jsonfile in jsonfiles:
+            with open(jsonfile, 'r') as f:
+                jsondata_list.append(json.load(f))
+        if len({d["renderParametersUrlTemplate"] for d in jsondata_list}) != 1:  # pragma: no cover
+            raise ValueError(
+                "Found tilepair files with disparate "
+                "renderParametersUrlTemplate values. "
+                "Maybe there are additional files "
+                "matching the outjson pattern?")
+        pairdata = [i for l in (d["neighborPairs"] for d in jsondata_list)
+                    for i in l]
+        jsondata = dict(jsondata_list[0], **{"neighborPairs": pairdata})
 
     if delete_json:
-        os.remove(outjson)
-    return jsondata
+        for jsonfile in jsonfiles:
+            os.remove(jsonfile)
+
+    return ((jsondata, jsonfiles) if return_jsonfiles and return_jsondata

```

Comment:
in the case of delete_json does it make sense to return paths to files that aren't there?

Reply:
Practically speaking I don't think so, but since return_jsonfiles defaults to False it seems like we should keep the outputs explicit.  Returning an empty list (i.e. just return the files you can expect to find) might work too, and I don't have a strong preference.